src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1578,64 +1522,47 @@ bool AdaptDispatch::_EraseScrollback()
+            // Clear everything after the viewport.
             const DWORD dwTotalAreaBelow = csbiex.dwSize.X * (csbiex.dwSize.Y - sHeight);
             const COORD coordBelowStartPosition = { 0, sHeight };
-            // We don't use the _EraseAreaHelper here because _EraseSingleLineDistanceHelper does it all in one operation
-            fSuccess = _EraseSingleLineDistanceHelper(coordBelowStartPosition, dwTotalAreaBelow, csbiex.wAttributes);
+            // Again we need to use the default attributes, hence standardFillAttrs is false.
+            fSuccess = _conApi->PrivateFillRegion(coordBelowStartPosition, dwTotalAreaBelow, L' ', false);
 
             if (fSuccess)
             {
-                // If there is a section B, clear it.
-                const COORD coordBottomRight = { csbiex.dwSize.X, coordBelowStartPosition.Y };
-                const COORD coordRightStartPosition = { sWidth, 0 };
-                if (coordBottomRight.X > coordRightStartPosition.X)
-                {
-                    // We use the Area helper here because the Line helper would
-                    //      erase the parts of the screen we want to keep too
-                    fSuccess = _EraseAreaHelper(coordRightStartPosition, coordBottomRight, csbiex.wAttributes);
-                }
+                // Move the viewport (CAN'T be done in one call with SetConsoleScreenBufferInfoEx, because legacy)
+                SMALL_RECT srNewViewport;
+                srNewViewport.Left = Screen.Left;
+                srNewViewport.Top = 0;
+                // SetConsoleWindowInfo uses an inclusive rect, while GetConsoleScreenBufferInfo is exclusive
+                srNewViewport.Right = Screen.Right - 1;
+                srNewViewport.Bottom = sHeight - 1;
+                fSuccess = !!_conApi->SetConsoleWindowInfo(true, &srNewViewport);
 
                 if (fSuccess)
                 {
-                    // Move the viewport (CAN'T be done in one call with SetConsoleScreenBufferInfoEx, because legacy)
-                    SMALL_RECT srNewViewport;
-                    srNewViewport.Left = 0;
-                    srNewViewport.Top = 0;
-                    // SetConsoleWindowInfo uses an inclusive rect, while GetConsoleScreenBufferInfo is exclusive
-                    srNewViewport.Right = sWidth - 1;
-                    srNewViewport.Bottom = sHeight - 1;
-                    fSuccess = !!_conApi->SetConsoleWindowInfo(true, &srNewViewport);
-
-                    if (fSuccess)
-                    {
-                        // Move the cursor to the same relative location.
-                        const COORD newCursor = { Cursor.X - Screen.Left, Cursor.Y - Screen.Top };
-                        fSuccess = !!_conApi->SetConsoleCursorPosition(newCursor);
-                    }
+                    // Move the cursor to the same relative location.
+                    const COORD newCursor = { Cursor.X, Cursor.Y - Screen.Top };
```

Comment:
Since we're setting the `srNewViewport.Left` to `Screen.Left`, should we use the viewport relative cursor position too?

```suggestion
                    const COORD newCursor = { Cursor.X - Screen.Left, Cursor.Y - Screen.Top };
```

Reply:
No, this is deliberate. The viewport has moved up from the original `Screen.Top` to 0, so we need to subtract `Screen.Top` from the original Y coordinate. The viewport's horizontal offset doesn't change though - hence `srNewViewport.Left` is equal to the original `Screen.Left` - and so we don't need to subtract anything from the X coordinate.

Reply:
Okay sounds good to me.